### PR TITLE
[SDK-445] Change client version header

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

This changes the already existing version header of the clients. However the previous version of the implementation did not follow the given scheme nor it provided the expected header values.

## Motivation and Context

This part of [SDK-445](https://gesundheitscloud.atlassian.net/browse/SDK-445) and depends on #131.
The backend is in need of not generic version header of the clients.

## How is it being implemented?

It pulls out the already existing interceptor, refactors it and aligns it to the given header schema. Also this PR changes slightly the clients, so that the header can separate between ingestion, jvm and android.

## How Has This Been Tested?

A new testset for the Interceptor had been added alongside with some changes in regards to ApiService.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sdk-core/src/main/java/care/data4life/sdk/network/NetworkingContract.kt [hunk 1]
```diff
@@ -123,18 +133,18 @@ interface NetworkingContract {
...
         const val HPSGC_DE = "sha256/3f81qEv2rjHvcrwof2egbKo5MjjSHaN/4DOl7R+pH0E="
         const val SMART4HEALTH_EU = "sha256/yPBKbgJMVnMeovGKbAtuz65sfy/gpDu0WTiuB8bE5G0="
-        const val REQUEST_TIMEOUT = 2
+        const val REQUEST_TIMEOUT: Long = 2
         const val HEADER_ALIAS = "gc_alias"
         const val HEADER_ACCESS_TOKEN = "access_token"
         const val HEADER_AUTHORIZATION = "Authorization"
         const val HEADER_BASIC_AUTH = "basic_auth"
-        const val HEADER_GC_SDK_VERSION = "GC-SDK-Version"
+        const val HEADER_GC_SDK_VERSION = "gc-sdk-version"
```
[the remark this answers is not in the document]
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/network/NetworkingContract.kt:141]: You are absolutely right - I will change it probably...glad you spotted it, thx.

[the remark this answers is not in the document]
Comment (alessioborraccino) [reply sdk-core/src/main/java/care/data4life/sdk/network/NetworkingContract.kt:141]: Requesting change

**Review by alessioborraccino — CHANGES_REQUESTED** [review]

File: sdk-core/src/main/java/care/data4life/sdk/network/NetworkingContract.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/network/NetworkingContract.kt:141]: done

File: sdk-ingestion/src/main/java/care/data4life/sdk/Data4LifeClient.java [hunk 2]
```diff
@@ -128,7 +131,18 @@ public static Data4LifeClient init(byte[] accessToken,
...
-        ApiService apiService = new ApiService(authorizationService, environment, clientId, DUMMY_CLIENT_SECRET, platform, networkConnectivityService, CLIENT_NAME, accessToken, DEBUG);
+        ApiService apiService = new ApiService(
+                authorizationService,
+                environment,
+                clientId,
+                DUMMY_CLIENT_SECRET,
+                platform,
+                networkConnectivityService,
+                NetworkingContract.Clients.INGESTION,
+                VERSION,
```
Comment (wmontwe) [inline sdk-ingestion/src/main/java/care/data4life/sdk/Data4LifeClient.java:142]: The version is now static, but it is controlled by a Git tag, Idealy it is injected into the codebase.

**Review by wmontwe — CHANGES_REQUESTED** [review]
Either we change the version to be dynamically controlled or we need to create a release checklist to update it on releases

File: sdk-ingestion/src/main/java/care/data4life/sdk/Data4LifeClient.java (hunk 2 above)
Comment (bitPogo) [reply sdk-ingestion/src/main/java/care/data4life/sdk/Data4LifeClient.java:142]: You are totally right - My question is though, how we get it similar done like in the Android client/make the BuildConfig available also for JVM?

Comment (bitPogo) [reply sdk-ingestion/src/main/java/care/data4life/sdk/Data4LifeClient.java:142]: As discussed in other place - I will provide a follow up to make this dynamic.

**Review by wmontwe — APPROVED** [review]
👍

**Pull request merged**